sdk/storage/storage-blob/src/ContainerClient.ts

```diff
@@ -548,6 +548,55 @@ export class ContainerClient extends StorageClient {
     });
   }
 
+  /**
+   * Iterates over blobs under the specified container.
+   *
+   * @example
+   * for await (const blob of containerClient.listBlobs()) {
+   *   console.log(`Container: ${blob.name}`);
+   * }
+   *
+   * @example
+   * let iter1 = containerClient.listBlobs();
+   * let i = 1;
+   * for await (const blob of iter1) {
+   *   console.log(`${i}: ${blob.name}`);
+   *   i++;
+   * }
+   *
+   * @example
+   * let iter2 = await containerClient.listBlobs();
+   * i = 1;
+   * let blobItem = await iter2.next();
+   * do {
+   *  console.log(`Blob ${i++}: ${blobItem.value.name}`);
+   *  blobItem = await iter2.next();
+   * } while (blobItem.value);
+   *
+   *
+   * @param {ContainerListBlobsSegmentOptions} [options]
+   * @returns {AsyncIterableIterator<Models.BlobItem>}
+   * @memberof ContainerClient
+   */
+  public async *listBlobs(
+    options: ContainerListBlobsSegmentOptions = {}
+  ): AsyncIterableIterator<Models.BlobItem> {
+    let marker = undefined;
+    const containerClient = this;
+    const aborter = !options || !options.abortSignal ? Aborter.none : options.abortSignal;
+    do {
+      const listBlobsResponse: Models.ContainerListBlobFlatSegmentResponse = await containerClient.listBlobFlatSegment(
```

Comment:
Do we need to specify the type `: Models.ContainerListBlobFlatSegmentResponse`? I feel we should relax linting to not require this as the type can be inferred from the method call.

Reply:
Hmmm, I see linter errors if the type is not specified.
Instead, I can declare the variable using let and reuse the variable, that way type will be inferred from the method call.